docs/_content/ontwikkelaars/algemeen/notificaties.md

```diff
@@ -100,3 +91,153 @@ Bronnen:
+
+projects.topics.create/delete/get/list
+
+### Abonnementen (subscriptions)
+
+Ondersteuning voor: GET (``list`` en ``retrieve``), POST, PUT, PATCH en DELETE
+
+**/api/v1/abonnementen[/{id}]**
+```javascript
+{
+  "url": "https://example.com/api/v1/api/v1/abonnementen/761a81", // Alleen-lezen
+  "id": "761a81", // Alleen-lezen, eventueel een UUID als onderliggende techniek dit toestaat
+  "bron": {  // De "bron" vorm het topic: <rsin>-<naam>
+    "rsin": "082096752011",
+    "naam": "zaken"
+  },
+  "versturenNaar": "https://example.com/api/v1/notificaties/ontvangen"  // Optioneel
+}
+```
+
+TODO: Waarom bron als rsin+naam en geen URL naar bron?
+
+*Gebaseerd op Google Cloud Pub/Sub*
+
+projects.subscriptions.create/get/list/delete/patch
+- id = name
+- '{rsin}-{bronNaam}' = topic
+- versturenNaar = pushConfig.pushEndpoint
+
+### Abonnementen bij een bron opvragen
+
+Ondersteuning voor: GET (``list``)
+
+**/api/v1/bronnen/{id}/abonnementen**
+```javascript
+[
+  {
+    "url": "https://example.com/api/v1/api/v1/abonnementen/761a81", // Alleen-lezen
+    "id": "761a81", // Alleen-lezen, eventueel een UUID als onderliggende techniek dit toestaat
+    "bron": {  // De "bron" vorm het topic: <rsin>-<naam>
+      "rsin": "082096752011",
+      "naam": "zaken"
+    },
+    "versturenNaar": "https://example.com/api/v1/notificaties/ontvangen"  // Optioneel
+  }
+]
+```
+
+*Gebaseerd op Google Cloud Pub/Sub*
+
+projects.topics.subscribers.list
+
+### Notificaties versturen
+
+Ondersteuning voor: POST
+
+**/api/v1/bronnen/{id}/notificaties**
+```javascript
+{
+  "bronUrl": "https://example.com/api/v1/zaken/d7a22"
+  "resource": "status",
+  "resourceUrl": "https://example.com/api/v1/statussen/721c9",
+  "actie": "create",  // "create", "update", etc. niet "read" of "list" want dat is geen verandering)
+  "aanmaakDatum": "2018-01-01T17:00:00Z",
+  "kenmerken": {
+    "zaaktype": "https://example.com/api/v1/zaaktypen/5aa5c"
+  }
+}
+```
+
+*Gebaseerd op Google Cloud Pub/Sub*
+
+projects.topics.publish
+
+### Notificaties ophalen (pull)
+
+Ondersteuning voor: GET (``list``)
+
+**/api/v1/abonnementen/{id}/notificaties**
+```javascript
+[
+  {
+    "id": "3aa9c5",
+    "bron": {
+      "rsin": "082096752011",
+      "naam": "zaken"
+    }
+    "bericht": {
+      "bronUrl": "https://example.com/api/v1/zaken/d7a22"
+      "resource": "status",
+      "resourceUrl": "https://example.com/api/v1/statussen/721c9",
+      "actie": "create",  // "create", "update", etc. niet "read" of "list" want dat is geen verandering)
+      "aanmaakDatum": "2018-01-01T17:00:00Z",
+      "kenmerken": {
+        "zaaktype": "https://example.com/api/v1/zaaktypen/5aa5c"
+      }
+    }
+  }
+]
+```
+
+*Gebaseerd op Google Cloud Pub/Sub*
+
+projects.subscriptions.pull
+- id = messageId
+- aanmaakDatum = publishTime
+- other attributes = data
+- kenmerken = attributes
+
+Also, automatically ACK messages, by calling subscribtions/acknowledge after the notifications are pulled.
```

Comment:
Hoe werkt dit? Houdt de NC bij welke notificaties zijn gepulled/gepushed naar een subscriber?

Reply:
> Dit was verhelderend: https://cloud.google.com/pubsub/docs/subscriber als het gaat om bevestigen van berichten. Je zegt dus dat als een bericht afgeleverd is (hetzij push hetzij pull) je het op Acknowledged zet.

Precies. Het was nog een work in progress, dus nog niet alles wat uitgeschreven.

Reply:
Ik heb dit stukje ook weggehaald nu en meer gefocust op webhooks, aangezien we van Google Pub/Sub af zijn.